### ISSUE TITLE
Loadbalanced environments

### ISSUE THREAD

**Issue by PerplexDaniel** [KevinJump/Our.Umbraco.MaintenanceMode#33]
Does this package work in loadbalanced environments, or any multi-server environment? For example, we regularly use environments with 1 back-end Umbraco server and 1+ front-end servers. If we were to switch on Maintenance Mode on the back-end Umbraco server, will that setting propagate to other servers that will then also switch to Maintenance Mode?

I performed a quick local test with 2 instances of the same website connecting to the same database and turning on Maintenance Mode on instance A does not turn on Maintenance Mode on instance B so I suppose this is not intended to work? It would be interesting if this could work, in theory Umbraco's [distributed cache refreshing mechanism](https://our.umbraco.com/documentation/reference/cache/) can be used to ping other servers to also switch to Maintenance Mode (and back out of it).

**Comment by karltynan** [comment]
This would be really useful as a feature. I wish I could contribute to a solution, but I am really not experienced in this area.

**Comment by craigagnew** [comment]
Having run into this issue on a recent implementation I wonder if it would be useful to store the maintenance mode status object in the database and therefore able to be accessed in a distributed/load balanced setup. I'm considering raising a PR with that in mind. @KevinJump what's your thoughts? 

My particular use case is a v10 site and I had a look at where I might target a PR for this - I feel like I want to PR into `v10/dev` however looking at the [comparison](https://github.com/Jumoo/Our.Umbraco.MaintenanceMode/compare/v10.1.1...v10/dev) with `v10.1.1` tag unless I'm mistaken it seems v14 work is now in there? Could you point me in the right direction?

**Comment by KevinJump** [comment]
Hi, 

i think v10/dev is the branch if you want to update the v10/11/12/13 versions of the code. (v14 one is out, but don't worry too much about that if you PR to the v10 one for your needs we can will port anything to v14)

https://github.com/Jumoo/Our.Umbraco.MaintenanceMode/tree/v10/dev

**Comment by craigagnew** [comment]
Hey Kevin, thanks for the speedy response :)

Excellent, I'll certainly try. Looking at the latest on that branch it looks like the Our.Umbraco.MaintenanceMode project has been bumped to .net8.0 and Umbraco v14. Please keep me right as it could be my compatibility knowledge is sketchy, I'm not sure that'll be able to be installed in lower versions?

**Comment by KevinJump** [comment]
yes 🤦  - fixed it now with v10/main is the latest v10 branch has the fixes (+1 for pascal casing) that went into the v10.1.1 release)

_(v10/dev is dead)_
